src/c++/simulator/my_types.hpp

```diff
@@ -0,0 +1,21 @@
+﻿#pragma once
+
+enum class Entity
+{
+	IParticle,
```

Comment:
It sounds strange having an enum value for an abstract entity. I would keep only for Concrete types.
What is the reasoning for this enum value?

Reply:
Or perhaps let's get rid of them all and try to get the type information from the abstractions using `dynamic_cast`. I'm not sure if one is better than the other, though, must do some testing

Reply:
Yes, I liked this option too, seems cleaner to me.